### PR TITLE
Publish A New Extension: Parrot-Translate

### PR DESCRIPTION
**Pull request by Haojen**
## Description

A Powerful and Easy to use translation, Support TTS and lowerCamelCase/ALL_UPPERCASE Copy mode, More features introduce can visit this project readme https://github.com/Haojen/raycast-Parrot

### Normal translate 
Screen Shot 2022-03-04 at 12 48 13 PM

### lowerCaseCopy translate result
lowerCopyMode

  ### UPPERCASE_COPY translate result
UppercastCopy

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed in the `media` folder

### CODE REVIEW

File: extensions/parrot-translate/src/parrot.tsx
Comment (pernielsentikaer) [inline extensions/parrot-translate/src/parrot.tsx:144]:
```suggestion
```
The [navigationTitle](https://developers.raycast.com/basics/prepare-an-extension-for-store#navigation-title) should not be set here in the root

File: extensions/parrot-translate/src/components.tsx
Comment (pernielsentikaer) [inline extensions/parrot-translate/src/components.tsx:54]:
```suggestion
                        title="Copy Result"
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Haojen 👋 

Thanks for your contribution 💯 

I have some feedback and suggestions ready for you 💪 

- When opening the extension, then there is a button saying “Chinese-Simplified” – if I click the button and search it, it's then searching for “Autodetect-Chinese” – EDIT: Now I can see what it does, maybe it should first be visible after search – kind of confusing as it is right now 😄 

- Missing an option to have autodetect in preferences for the “First language”

- Instead of “Automatic Paste” in preferences, then you should add [Action.Paste](https://developers.raycast.com/api-reference/user-interface/actions#action.paste) instead 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by pernielsentikaer** [comment]
Hi @Haojen 👋 

Please remember that one PR is. only for one extension, could you remove it from this PR?

On a sidenote, did you see that we already have a [speedtest extension](https://www.raycast.com/tonka3000/speedtest) in the store?

**Comment by Haojen** [comment]
@pernielsentikaer  yeah, sorry about that. I developer another Extension, I forget this PR does not merge. I will revert. I searched and checked, but I really don't see this speedtest extension.  so so so sad.

**Comment by Haojen** [comment]
@pernielsentikaer I 13 days ago. this too fast happened, I can't always check has the too same extension. :(

File: extensions/parrot-translate/src/components.tsx
Comment (pernielsentikaer) [reply extensions/parrot-translate/src/components.tsx:54]: You need to change this too :)

File: extensions/parrot-translate/src/parrot.tsx
Comment (pernielsentikaer) [inline extensions/parrot-translate/src/parrot.tsx:138]:
```suggestion
            searchBarPlaceholder={"Translate to"}
            throttle={true}
```

File: extensions/parrot-translate/src/parrot.tsx
Comment (pernielsentikaer) [inline extensions/parrot-translate/src/parrot.tsx:55]:
```suggestion
        requestYoudaoAPI(inputState, translateTargetLanguage.languageId).then((res) => {
            const resData: ITranslateResult = res.data
    
            const [a, b] = resData.l.split("2") // en2zh
    
            if (a === b) {
                updateTranslateTargetLanguage(a === preferences.lang1 ? defaultLanguage2 : defaultLanguage1)
                return
            }
    
            updateLoadingState(false)
            fetchResultStateCode = res.data.errorCode
            updateTranslateResultState(reformatTranslateResult(resData))
    
            updateTranslateFromLanguageState(getItemFromLanguageList(a))
            updateCurrentTargetLanguageState(getItemFromLanguageList(b))
        })
```
With throttle on the list, this shouldn't be necessary

File: extensions/parrot-translate/src/components.tsx
Comment (pernielsentikaer) [inline extensions/parrot-translate/src/components.tsx:48]:
```suggestion
```
Redundant

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Haojen 👋 

I checked the latest update, and I have some feedback for you to address 😄 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

File: extensions/parrot-translate/src/parrot.tsx
Comment (Haojen) [reply extensions/parrot-translate/src/parrot.tsx:138]: The throttle API does not a good choice. here is why: https://github.com/raycast/extensions/issues/766

File: extensions/parrot-translate/src/components.tsx
Comment (Haojen) [reply extensions/parrot-translate/src/components.tsx:48]: the title key content and value key content are not the same when the translated resulting length is too long. so I need to keep the title and value key.

Comment (Haojen) [reply extensions/parrot-translate/src/components.tsx:48]: image

The Title is: `Copy All`, The value key actually is `How are you；How Do You Do；Are You OK`

Comment (pernielsentikaer) [reply extensions/parrot-translate/src/components.tsx:48]: Ah, fair!

**Comment by pernielsentikaer** [comment]
Hi @Haojen 👋 

There is a problem now that throttle is not used, see below - maybe you can find a way to have them both, as in my last video? Just added throttle:

Before:

https://user-images.githubusercontent.com/38033107/158200354-6cb9eac5-8fa3-4feb-9a8c-5a7508c93621.mp4

After:

https://user-images.githubusercontent.com/38033107/158200383-088c0168-4897-4a39-a48f-dfc34f9bc7c7.mp4

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Haojen 👋 

Sorry for the longer response time 😄 

I tried to start all over and installing it again, it selected this:

wTGyO4TOarZj6Dxn

However, I end up with this result:

JVNQriVtkYRj4CXo

It's the same when I change to another language, it seems like it's stuck with Chinese somehow.

Regarding the flicker on the details button, I would like it to be fixed before we can merge also, it's not giving a good UX

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by Haojen** [comment]
> Hi @Haojen 👋
> 
> Sorry for the longer response time 😄
> 
> I tried to start all over and installing it again, it selected this:
> 
> wTGyO4TOarZj6Dxn
> 
> However, I end up with this result:
> 
> JVNQriVtkYRj4CXo
> 
> It's the same when I change to another language, it seems like it's stuck with Chinese somehow.
> 
> Regarding the flicker on the details button, I would like it to be fixed before we can merge also, it's not giving a good UX
> 
> Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

1. Fixed always is Chinese language bug
2. Fixed action flash bug
3. Improved translate speed
4. Improved stability and detail optimization

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Haojen 👋 

Thanks for the update, it does indeed work better now 🔥 

We're now adding support for [categories](https://developers.raycast.com/basics/prepare-an-extension-for-store#categories), [screenshots](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) and [changelog](https://developers.raycast.com/basics/prepare-an-extension-for-store#version-history) – do you have time for adding this too before final review?

**Comment by Haojen** [comment]
@pernielsentikaer Good to hear, I later I'll add it later

**Comment by Haojen** [comment]
@pernielsentikaer  I added categories, screenshots and changelog.  waiting so long : )

File: extensions/parrot-translate/package.json
Comment (pernielsentikaer) [inline extensions/parrot-translate/package.json:273]:
```suggestion
        "@raycast/api": "^1.31.0",
```

File: extensions/parrot-translate/src/parrot.tsx
Comment (pernielsentikaer) [inline extensions/parrot-translate/src/parrot.tsx:167]:
```suggestion
            searchBarPlaceholder={"Translate text"}
```

File: extensions/parrot-translate/src/parrot.tsx
Comment (pernielsentikaer) [inline extensions/parrot-translate/src/parrot.tsx:175]:
```suggestion
            <List.EmptyView
                icon="parrot-icon-small.png"
                title="Type something to get started"
            />
            <ListDetail />
```
Add the attached icon to assets 😄

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Haojen 👋 

Thanks for the update, much appreciated 🙏 

I think we're about to be ready, I have the last feedback for you before merging 💪


This is the image for assets folder:
parrot-icon-small 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Review by pernielsentikaer — APPROVED** [review]
Hi @Haojen 👋 

Awesome, let's get the first version up and running and do a version 2 with the rest as we spoke about 💪

**Pull request merged**
